File: ui/src/test/java/com/kotcrab/vis/ui/test/LesserThanValidatorTest.java

package com.kotcrab.vis.ui.test;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import com.kotcrab.vis.ui.util.Validators.LesserThanValidator;

import org.junit.Test;

public class LesserThanValidatorTest {
    @Test
    public void testValidateInput() throws Exception {
        LesserThanValidator validator = new LesserThanValidator(5);
        LesserThanValidator validatorEquals = new LesserThanValidator(5, true);

        assertFalse(validator.validateInput("A"));
        assertFalse(validator.validateInput(""));
        assertFalse(validatorEquals.validateInput("A"));
        assertFalse(validatorEquals.validateInput(""));

        assertFalse(validator.validateInput("5"));
        assertTrue(validator.validateInput("4"));
        assertTrue(validatorEquals.validateInput("5"));
        assertTrue(validatorEquals.validateInput("4"));

        assertFalse(validator.validateInput(String.valueOf(Float.MAX_VALUE)));
        assertFalse(validatorEquals.validateInput(String.valueOf(Float.MAX_VALUE)));

        assertTrue(validator.validateInput(String.valueOf(Float.MIN_VALUE)));
        assertTrue(validatorEquals.validateInput(String.valueOf(Float.MIN_VALUE)));
    }
}

File: ui/src/test/java/com/kotcrab/vis/ui/test/LesserThanValidatorTest.kt
package com.kotcrab.vis.ui.test

import com.kotcrab.vis.ui.util.Validators.LesserThanValidator
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class LesserThanValidatorTest {

    @Test
    fun testValidateInput() {
        val validator = LesserThanValidator(5F)
        val validatorEquals = LesserThanValidator(5F, true)

        assertFalse(validator.validateInput("A"))
        assertFalse(validator.validateInput(""))
        assertFalse(validator.validateInput("5"))
        assertTrue(validator.validateInput("4"))
        assertFalse(validator.validateInput(Float.MAX_VALUE.toString()))
        assertTrue(validator.validateInput(Float.MIN_VALUE.toString()))


        assertFalse(validatorEquals.validateInput("A"))
        assertFalse(validatorEquals.validateInput(""))
        assertTrue(validatorEquals.validateInput("5"))
        assertTrue(validatorEquals.validateInput("4"))
        assertFalse(validatorEquals.validateInput(Float.MAX_VALUE.toString()))
        assertTrue(validatorEquals.validateInput(Float.MIN_VALUE.toString()))
    }
}
